crates/iceberg/src/expr/visitors/row_group_metrics_evaluator.rs

```diff
@@ -0,0 +1,523 @@
+    ) -> Result<bool> {
+        if row_group_metadata.num_rows() == 0 {
+            return ROW_GROUP_CANT_MATCH;
+        }
+
+        let mut evaluator = Self::new(row_group_metadata, field_id_map, snapshot_schema);
+
+        visit(&mut evaluator, filter)
+    }
+
+    fn stats_for_field_id(&self, field_id: i32) -> Option<&Statistics> {
+        let parquet_column_index = *self
+            .iceberg_field_id_to_parquet_column_index
+            .get(&field_id)?;
+        self.row_group_metadata
+            .column(parquet_column_index)
+            .statistics()
+    }
+
+    fn null_count(&self, field_id: i32) -> Option<u64> {
+        self.stats_for_field_id(field_id)
+            .map(|stats| stats.null_count())
+    }
+
+    fn value_count(&self) -> u64 {
+        self.row_group_metadata.num_rows() as u64
+    }
+
+    fn contains_nulls_only(&self, field_id: i32) -> bool {
+        let null_count = self.null_count(field_id);
+        let value_count = self.value_count();
+
+        null_count.is_some() && null_count == Some(value_count)
+    }
+
+    fn may_contain_null(&self, field_id: i32) -> bool {
+        if let Some(null_count) = self.null_count(field_id) {
+            null_count > 0
+        } else {
+            true
+        }
+    }
+
+    fn stats_and_type_for_field_id(
+        &self,
+        field_id: i32,
+    ) -> Result<Option<(&Statistics, PrimitiveType)>> {
+        let Some(stats) = self.stats_for_field_id(field_id) else {
+            // No statistics for column
+            return Ok(None);
+        };
+
+        let Some(field) = self.snapshot_schema.field_by_id(field_id) else {
+            return Err(Error::new(
+                ErrorKind::Unexpected,
+                format!(
+                    "Could not find a field with id '{}' in the snapshot schema",
+                    &field_id
+                ),
+            ));
+        };
+
+        let Some(primitive_type) = field.field_type.as_primitive_type() else {
+            return Err(Error::new(
+                ErrorKind::Unexpected,
+                format!(
+                    "Could not determine the PrimitiveType for field id '{}'",
+                    &field_id
+                ),
+            ));
+        };
+
+        Ok(Some((stats, primitive_type.clone())))
+    }
+
+    fn min_value(&self, field_id: i32) -> Result<Option<Datum>> {
+        let Some((stats, primitive_type)) = self.stats_and_type_for_field_id(field_id)? else {
+            return Ok(None);
+        };
+
+        Ok(Some(Datum::try_from_bytes(
+            stats.min_bytes(),
+            primitive_type,
+        )?))
+    }
+
+    fn max_value(&self, field_id: i32) -> Result<Option<Datum>> {
+        let Some((stats, primitive_type)) = self.stats_and_type_for_field_id(field_id)? else {
+            return Ok(None);
+        };
+
+        Ok(Some(Datum::try_from_bytes(
+            stats.max_bytes(),
+            primitive_type,
+        )?))
+    }
```

Comment:
I've updated (not via a force-push 😅 ) the PR to make use of `min` and `max` rather than `min_bytes` and `max_bytes`, as well as using `has_min_max_set`. I've covered all but the most obscure primitive types, as we've done similarly in other places.

> min/max may not be exact, they maybe truncated for binary values, we should check min_is_exact/max_is_exact method to ensure that.

What should the behaviour be when `min_is_exact() == false` /  `max_is_exact() == false`?